opencl-aot/source/main.cpp

```diff
@@ -264,6 +282,23 @@ int main(int Argc, char *Argv[]) {
           clEnumVal(gpu, "Intel(R) Processor Graphics device"),
           clEnumVal(fpga_fast_emu,
                     "Intel(R) FPGA Emulation Platform for OpenCL(TM) device")));
+
+  // Group options which only used with FPGA emulator
+  enum Commands { build, compile, link };
+  cl::opt<Commands> OptCommand(
+      "cmd", cl::desc("Command"),
+      cl::values(
+          clEnumVal(
+              build,
+              "Build executable IR from OpenCL source or OpenCL .obj file"),
+          clEnumVal(compile, "Compile from OpenCL source to .obj"),
+          clEnumVal(link, "Link OpenCL .obj file(s)")));
+  cl::opt<std::string> OptInput("input", cl::desc("Input OpenCL C file"));
+  cl::opt<std::string> OptIR("ir", cl::desc("Output file"));
+  cl::opt<std::string> OptSPIRV("spv", cl::desc("Input SPIR-V file"));
+  cl::opt<std::string> OptBinaries("binary",
+                                   cl::desc("Input OpenCL .obj files to link"));
```

Comment:
As we discussed offline, these options are only needed for compatibility with some other Intel-provided tool.
Please put a comment describing that and add `TODO`/`FIXME` to delete this option once transition to a regular `-o` and regular input argument is complete.

Moreover, I suggest that for inputs consistency checking we use actual filetypes which we detect by looking into file content like it is already done:
https://github.com/intel/llvm/blob/41443127d775c6e09686dd11dbffcf9186c44e01/opencl-aot/source/main.cpp#L146-L161

And I also suggest to use [aliases](https://llvm.org/docs/CommandLine.html#argument-aliases) to leave or logic to work with regular input/output arguments and declare new arguments as aliases to the existing ones